docs/smart-contracts/guides/transactions/simulateTransaction-Deep-Dive.mdx

```diff
@@ -0,0 +1,403 @@
+}
+```
+
+Here is an example for handling restoration using the `restorePreamble` to restore archived data:
+
+```typescript
+/**
+ * Simulates a restoration transaction to determine if restoration is needed.
+ * This function first checks the ledger entry for the given WASM hash. If the entry is found and has expired,
+ * it attempts a restoration. If the entry hasn't expired yet but the TTL needs extension, it proceeds with TTL extension.
+ * @param contract - The address of the contract to check
+ * @param txParams - The transaction parameters including account and signer.
+ * @returns A promise that resolves to a simulation response, indicating whether restoration or TTL extension is needed.
+ */
+export async function simulateRestorationIfNeeded(
+  contract: ContractAddress,
+  txParams: TxParams,
+): Promise<
+  SorobanRpc.Api.SimulateTransactionRestoreResponse | string | undefined
+> {
+  try {
+    const account = await config.rpc.getAccount(txParams.account.accountId());
+    const contract = new Contract(contract);
+    //const ledgerKey = getLedgerKeyWasmId(wasmHash);
+    const ledgerKey = contract.getFootprint();
+    const response = await config.rpc.getLedgerEntries(...[ledgerKey]);
+    if (
+      response.entries &&
+      response.entries.length > 0 &&
+      response.entries[0].liveUntilLedgerSeq
+    ) {
+      const expirationLedger = response.entries[0].liveUntilLedgerSeq;
+      const desiredLedgerSeq = response.latestLedger + 500000;
+      let extendLedgers = desiredLedgerSeq - expirationLedger;
+      if (extendLedgers < 10000) {
+        extendLedgers = 10000;
+      }
+      console.log("Expiration:", expirationLedger);
+      console.log("Desired TTL:", desiredLedgerSeq);
+      const sorobanData = new SorobanDataBuilder()
+        .setReadWrite([ledgerKey])
+        .build();
+      const restoreTx = new TransactionBuilder(
+        account,
+        txParams.txBuilderOptions,
+      )
+        .setSorobanData(sorobanData)
+        .addOperation(Operation.restoreFootprint({})) // The actual restore operation
+        .build();
+      // Simulate a transaction with a restoration operation to check if it's necessary
+
+      const restorationSimulation: SorobanRpc.Api.SimulateTransactionResponse =
+        await config.rpc.simulateTransaction(restoreTx);
+
+      //check if restore is necessary. this code also checks if the simulation was successful.
+      const restoreNeeded = SorobanRpc.Api.isSimulationRestore(
+        restorationSimulation,
+      );
+      console.log(`restoration needed: ${restoreNeeded}`);
+      // Check if the simulation indicates a need for restoration
+      if (restoreNeeded) {
+        return restorationSimulation as SorobanRpc.Api.SimulateTransactionRestoreResponse;
+      } else {
+        console.log("No restoration needed., bumping the ttl.");
+        const account1 = await config.rpc.getAccount(
```

Comment:
`config`?

Reply:
> `config`?

It is basically my dotenv, but I've changed now all instances of using the config object for the correct methods, for example we now instantiate an RpcServer at the start of the function.